modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -717,7 +717,14 @@ private CompletableFuture<List<BinaryRow>> continueScanBatchRetrieval(
             List<BinaryRow> batchRows
     ) {
         if (batchRows.size() < batchCount && cursor.hasNext()) {
-            return resolveAndCheckReadCompatibility(cursor.next(), txId)
+            ReadResult readResult = cursor.next();
+
+            HybridTimestamp newestCommitTimestamp = readResult.newestCommitTimestamp();
```

Comment:
No, the logic is executed asynchronously. That means the cursor might move ahead when we try to find a candidate.

Reply:
I didn't get your point. Could you please elaborate it?

Reply:
It should be executed synchronously to get a committed row from the cursor. We cannot get the row after the resolution because the resolution is asynchronously completed.

Reply:
It's still not clear.